File: backend/tests/vehicleNumber.test.js
'use strict';

const { normalizeVehicleNumber, isValidVehicleNumber } = require('../src/utils/vehicleNumber');

describe('vehicleNumber utils', () => {
  test('normalizes spaces, dashes and case', () => {
    expect(normalizeVehicleNumber('mh12-ab 1234')).toBe('MH12AB1234');
    expect(normalizeVehicleNumber('  MH 14 XY 5678  ')).toBe('MH14XY5678');
    expect(normalizeVehicleNumber('mh.12.ab.1234')).toBe('MH12AB1234');
    expect(normalizeVehicleNumber(null)).toBe('');
  });

  test('validates regex', () => {
    expect(isValidVehicleNumber('MH12AB1234')).toBe(true);
    expect(isValidVehicleNumber('ABCD')).toBe(true);
    expect(isValidVehicleNumber('AB')).toBe(false);
    expect(isValidVehicleNumber('TOOLONG12345678')).toBe(false);
    expect(isValidVehicleNumber('mh12ab1234')).toBe(false); // expects uppercase
  });
});

File: mobile/test/vehicle_number_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:vehicle_survey/utils/vehicle_number.dart';

void main() {
  group('VehicleNumber.normalize', () {
    test('strips spaces, dashes, dots and uppercases', () {
      expect(VehicleNumber.normalize('mh12-ab 1234'), 'MH12AB1234');
      expect(VehicleNumber.normalize('  MH 14 XY 5678  '), 'MH14XY5678');
      expect(VehicleNumber.normalize('mh.12.ab.1234'), 'MH12AB1234');
      expect(VehicleNumber.normalize(null), '');
    });

    test('isValid checks regex', () {
      expect(VehicleNumber.isValid('MH12AB1234'), isTrue);
      expect(VehicleNumber.isValid('ABCD'), isTrue);
      expect(VehicleNumber.isValid('AB'), isFalse);
      expect(VehicleNumber.isValid('TOOLONG12345678'), isFalse);
    });
  });
}
